src/textual/widgets/_directory_tree.py

```diff
@@ -91,18 +94,58 @@ def __init__(
         classes: str | None = None,
         disabled: bool = False,
     ) -> None:
-        str_path = os.fspath(path)
-        self.path = str_path
+        """Initialise the directory tree.
+
+        Args:
+            path: Path to directory.
+            name: The name of the widget, or None for no name.
+            id: The ID of the widget in the DOM, or None for no ID.
+            classes: A space-separated list of classes, or None for no classes.
+            disabled: Whether the directory tree is disabled or not.
+        """
+        self.path = path
         super().__init__(
-            str_path,
-            data=DirEntry(str_path, True),
+            str(path),
+            data=DirEntry(Path(path)),
             name=name,
             id=id,
             classes=classes,
             disabled=disabled,
         )
 
-    def process_label(self, label: TextType):
+    def reload(self) -> None:
+        """Reload the `DirectoryTree` contents."""
+        self.reset(str(self.path), DirEntry(Path(self.path)))
+        self._load_directory(self.root)
+
+    def validate_path(self, path: str | Path) -> Path:
+        """Ensure that the path is of the `Path` type.
+
+        Args:
+            path: The path to validate.
+
+        Returns:
+            The validated Path value.
+
+        Note:
+            The result will always be a Python `Path` object, regardless of
+            the value given.
+        """
+        return Path(path)
+
+    def watch_path(self, new_path: str | Path) -> None:
+        """Watch for changes to the `path` of the directory tree.
+
+        Args:
+            new_path: The new path being set.
+
+        If the path is changed the directory tree will be repopulated using
+        the new value as the root.
+        """
+        self.path = Path(new_path)
```

Comment:
I'm confused about this. If we have a validator, won't this always be a `Path` ?

And if we're watching path, do we need to set path again?

Reply:
Nah, good catch, looks like a complete thinko on my part. 100% no reason ti assign it again.